packages/jsii/lib/assembler.ts

```diff
@@ -3361,3 +3366,7 @@ function loadAndRenderReadme(readmePath: string, projectRoot: string) {
       .join('\n'),
   };
 }
+
+function compressJsii(assembly: spec.Assembly): boolean {
+  return assembly.metadata?.jsii?.compress ?? false;
+}
```

Comment:
It feels backwards to me that the place where you source the "should the assembly be compressed" setting is... in the assembly itself. I don't think that setting should be forwarded to the assembly at all (see previous comment about how this shouldn't be buried under metadata).

```suggestion
function assemblyCompressionEnabled(assembly: spec.Assembly): boolean {
  return assembly.metadata?.jsii?.compress == true;
}
```